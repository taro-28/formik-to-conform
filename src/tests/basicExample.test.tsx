import { convert } from "..";
import { test, expect, describe, afterEach } from "vitest";
import { readFileSync } from "node:fs";
import path from "node:path";
import type { ReactNode } from "react";
import { BasicExample as FormikBasicExample } from "./formik.basicExample";
import { BasicExample as ConformBasicExample } from "./conform.basicExample";
import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
test("convert", () => {
  const formikFile = readFileSync(
    path.join(__dirname, "formik.basicExample.tsx"),
    "utf-8",
  );
  const conformFile = readFileSync(
    path.join(__dirname, "conform.basicExample.tsx"),
    "utf-8",
  );
  const result = convert(formikFile);
  expect(result).toEqual(conformFile);
});

describe("compornent test", () => {
  afterEach(() => {
    cleanup();
  });
  test.each<{ name: string; Component: () => ReactNode }>([
    {
      name: "formik",
      Component: FormikBasicExample,
    },
    {
      name: "conform",
      Component: ConformBasicExample,
    },
  ])("render $name", ({ Component }) => {
    render(<Component />);

    expect(
      screen.getByRole<HTMLInputElement>("textbox", { name: "Name" }).value,
    ).toBe("jared");
  });
});
