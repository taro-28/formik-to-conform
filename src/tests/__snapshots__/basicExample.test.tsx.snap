// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compornent test > render 'conform' 1`] = `
<div>
  <form>
    <input
      form="«r0»"
      id="«r0»-name"
      name="name"
      type="text"
      value="jared"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`compornent test > render 'formik' 1`] = `
<div>
  <form>
    <input
      name="name"
      value="jared"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
